src/libstore/dummy-store.cc

```diff
@@ -18,9 +18,12 @@ struct DummyStoreConfig : virtual StoreConfig {
 
 struct DummyStore : public virtual DummyStoreConfig, public virtual Store
 {
-    DummyStore(const std::string scheme, const std::string uri, const Params & params)
+    DummyStore(std::string_view scheme, std::optional<std::string_view> authority, const Params & params)
         : DummyStore(params)
-    { }
+    {
+        if (authority)
+            warn("`dummy://` store URIs should not contain an authority part. This will become an error in future versions of Nix");
+    }
```

Comment:
I think we can directly make this an error. Not really worth a proper deprecation

Reply:
So re the `std::optional`, I am thinking that a lot of the `value_or("")` I did is kinda sketchy. If you think it is OK to go straight to hard errors, then I think we can get *more* value out of the `std::optional`, because it makes it easier to remember to handle the case.

Reply:
The version I just pushed has a bunch more checking that things are empty or non-empty. The only `value_or("")` left is for `file://` which is the same thing as `file:///`. (Not sure we want that one either.)